packages/web-components/src/components/va-search-filter/va-search-filter.tsx

```diff
@@ -0,0 +1,346 @@
+      console.error('filterOptions must be an array');
+      return null;
+    }
+
+  /**
+   * Checks that the top-level facet:
+   * - Has a valid label (string and non-empty)
+   * - Has a valid ID (number or string)
+   * - Has a category array
+   * Checks that each facet category item:
+   * - Has a valid label (string and non-empty)
+   * - Has a valid ID (number or string)
+   */
+    const validOptions = options.every(option => {
+      const hasValidLabel = typeof option.label === 'string' && option.label.length > 0;
+      const hasValidCategory = Array.isArray(option.category) && option.category.length > 0 && option.category.every(cat => {
+        const hasValidCatLabel = typeof cat.label === 'string' && cat.label.length > 0;
+        const hasValidCatId = typeof cat.id === 'number' || typeof cat.id === 'string';
+        return hasValidCatLabel && hasValidCatId;
+      });
+      const hasValidId = typeof option.id === 'number' || typeof option.id === 'string';
+      return hasValidLabel && hasValidCategory && hasValidId;
+    });
+
+    return validOptions ? options : null;
+  }
+
+  /**
+   * This method is invoked once before the component is first rendered.
+   * Its main purpose is to validate and set initial active filter counts.
+   */
+  componentWillLoad() {
+    if (!this.filterOptions?.length) return;
+
+    const validatedOptions = VaSearchFilter.validateFilterOptions(this.filterOptions);
+
+    if (validatedOptions) {
+      // update filterOptions with validated options and initial active filter counts.
+      this.totalActiveFilters = this.getTotalActiveFilters();
+      this.filterOptions = validatedOptions.map((facet) => {
+        return {
+          ...facet,
+          activeFiltersCount: VaSearchFilter.getTotalActiveFiltersByFacet(facet),
+        };
+      }); 
+    } else {
+      this.filterOptions = [];
+    }
+  }
+
+  render() {
+    const {
+      header,
+      filterOptions,
+      isDesktop,
+      totalActiveFilters,
+      handleFilterChange,
+      handleClearAllFilters,
+      handleApplyFilters,
+    } = this;
+
+    const filterButtons = (
+      <div id="filter-buttons">
+        <va-button
+          onClick={handleApplyFilters}
+          text="Apply filters"
+          full-width
+        />
+        <va-button
+          onClick={handleClearAllFilters}
+          text="Clear all filters"
+          secondary
+          full-width
+        />
+      </div>
+    );
+
+    if (isDesktop) {
+      return (
+        <Host>
+          <h2 id="header">{header}{totalActiveFilters > 0 ? ` (${totalActiveFilters})` : ''}</h2>
+          {filterOptions.length > 0 && (
+            <va-accordion class="va-search-filter__accordion">
+              {filterOptions.map((facet: FilterFacet) => (
+                <va-accordion-item 
+                  header={facet.label + (facet.activeFiltersCount > 0 ? ` (${facet.activeFiltersCount})` : '')} 
+                  key={facet.id} 
+                  open
```

Comment:
There is a bug with va-accordion-item that requires two clicks before it will close when `open` is set. I created a ticket for that here:

- https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/4010